### PR TITLE
changed attributes populated from module-catalog instead of module-ca…

### LINKED ISSUE #26002: When category selected from top level navigation, attributes populated from module-catalog-search instead of module-catalog
### Preconditions (*)

1. Magento ver 2.3.3
2. PHP  7.2.24-0
3. Ubuntu 18.04.1

### Steps to reproduce (*)

1. Install Magento 
2. Add a log entry in /var/www/html/magento23/vendor/magento/module-catalog-search/Model/Layer/Filter/Attribute.php near line 81 (first line in _getItemsData() method
3. In the default LUMA store select Men -> Tops -> Jackets

### Expected result (*)

1. I expect the code from /var/www/html/magento23/vendor/magento/module-catalog/Model/Layer/Filter/Attribute.php to execute to populate the left hand navigation

### Actual result (*)

1. The code from /var/www/html/magento23/vendor/magento/module-catalog-search/Model/Layer/Filter/Attribute.php is executed instead.  I expect a clear demarcation of catalog and catalog-search functionality. When you do a search module-catalog-search should be executed and when you select a category module-catalog should be executed.

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
…talog-search

### Fixed Issues (if relevant)

1. fixes magento/magento2#26002: When category selected from top level navigation, attributes populated from module-catalog-search instead of module-catalog 

### Manual testing scenarios (*)

1.Install Magento
2.Add a log entry in /var/www/html/magento23/vendor/magento/module-catalog-search/Model/Layer/Filter/Attribute.php near line 81 (first line in _getItemsData() method
3.In the default LUMA store select Men -> Tops -> Jackets

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by srsathish92** [comment]
@ProkopovVitaliy Given fix not working for swatches with layered navigation. I have tried the same earlier in this PR #28058. can you please check failure functional test, May be it will help.

Additional Info: Its throw errors on lib/internal/Magento/Framework/Data/Collection.php: Line No: 408

**Comment by ProkopovVitaliy** [comment]
@srsathish92 Thanks for the info, I didn't pay attention to it.

I will check and fix it.

File: app/code/Magento/CatalogSearch/etc/di.xml [hunk 1]
```diff
@@ -135,16 +135,6 @@
     <type name="Magento\Catalog\Model\Layer\Search\CollectionFilter">
         <plugin name="searchQuery" type="Magento\CatalogSearch\Model\Layer\Search\Plugin\CollectionFilter" />
     </type>
-    <virtualType name="categoryFilterList" type="Magento\Catalog\Model\Layer\FilterList">
```
Comment (orlangur) [inline app/code/Magento/CatalogSearch/etc/di.xml:138]: What was the purpose of such declaration at all?

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by engcom-Charlie** [comment]
Hi @ProkopovVitaliy, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**